VultisigApp/VultisigApp/Services/Rates/RateProvider.swift

```diff
@@ -74,10 +74,35 @@ final class RateProvider {
         return balance.formatToFiat(includeCurrencySymbol: true)
     }
 
+    // Cache of recently looked up rates to avoid repeated lookups
+    private var rateCache: [String: Rate?] = [:]
+    
     func rate(for coin: Coin, currency: SettingsCurrency = .current) -> Rate? {
         let cryptoId = RateProvider.cryptoId(for: coin)
-        let identifier = Rate.identifier(fiat: currency.rawValue, crypto: cryptoId.id)
-        return rates.first(where: { $0.id == identifier })
+        let identifier = Rate.identifier(fiat: currency.rawValue.lowercased(), crypto: cryptoId.id.lowercased())
```

Comment:
This is a cache without timeout , which means as long as the app is open , the price will never updated